sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -270,6 +271,19 @@ class SessionCatalog(
   }
 
   /**
+   * Retrieve the metadata of an existing permanent table/view. If no database is specified,
+   * assume the table/view is in the current database. If the specified table/view is not found
+   * in the database then a [[NoSuchTableException]] is thrown.
+   */
+  def getNonTempTableMetadata(name: TableIdentifier): CatalogTable = {
```

Comment:
I don't think `getTableMetadata` should handle temp view, maybe we can fix it in this PR?